feg/gateway/services/n7_n40_proxy/servicers/session_controller.go

```diff
@@ -94,6 +95,22 @@ func (srv *CentralSessionController) TerminateSession(
 	ctx context.Context,
 	request *protos.SessionTerminateRequest,
 ) (*protos.SessionTerminateResponse, error) {
-
-	return (&protos.UnimplementedCentralSessionControllerServer{}).TerminateSession(ctx, request)
+	glog.V(2).Info("Deleting session")
+	if err := validateSessionTerminateRequest(request); err != nil {
+		return nil, err
+	}
+	smPolicyId := n7.GetSmPolicyId(request.GetTgppCtx())
+	if len(smPolicyId) == 0 {
+		return nil, fmt.Errorf("SmPolicyId not found in the SessionTerminateResponse TgppCtx")
+	}
+	reqBody := n7.GetSmPolicyDeleteReqBody(request)
+	err := srv.sendSmPolicyDelete(smPolicyId, reqBody)
+	metrics.ReportDeleteSmPolicy(err)
+	if err != nil {
+		return nil, err
+	}
+	return &protos.SessionTerminateResponse{
+		Sid:       request.GetCommonContext().GetSid().GetId(),
+		SessionId: request.SessionId,
+	}, nil
```

Comment:
in my o_pinion, i would add a glog.Error in any error returned by grpc endpoint. (not just in terminate, but in all session_controller.go endpoints

Reply:
Done